public/js/draw/draw.js

```diff
@@ -389,3 +390,107 @@ function regionClicked(e) {
         prevY = position.y;
     }
 }
+
+
+/**
+ * Parse SVG elements into a JSON list.
+ */
+function convertSvgToJson(gId) {
+    'use strict';
+
+    var texts = [];
+    var shapes = [];
+    var paths = [];
+    var svgElements = document.getElementById('mySVG').children;
+    for (var i = 0; i < svgElements.length; i++) {
+        if (svgElements[i].tagName === 'g' && svgElements[i].id !== 'regions') {
+            var textObj = '';
+            var gText = svgElements[i].querySelector('text');
+            if (gText !== null) {
+                textObj = {
+                    'graph'     : parseInt(gId),
+                    'rId'       : '',
+                    'pos'       : [ parseFloat(gText.getAttribute('x')),
+                                    parseFloat(gText.getAttribute('y'))
+                                    ],
+                    'text'      : gText.textContent,
+                    'align'     : 'begin',
+                    'fill'      : ''
+                };
+               texts.push(textObj);
+            }
+
+            var gRect = svgElements[i].querySelector('rect');
+            shapes.push({
+                    'graph'     : parseInt(gId),
+                    'id_'       : '',
+                    'pos'       : [ parseFloat(gRect.getAttribute('x')),
+                                    parseFloat(gRect.getAttribute('y'))
+                                    ],
+                    'width'     : parseFloat(gRect.getAttribute('width')),
+                    'height'    : parseFloat(gRect.getAttribute('height')),
+                    'fill'      : 'none',
+                    'stroke'    : '',
+                    'text'      : [textObj],
+                    'tolerance' : 9,
+                    'type_'     : 'Node'
+            });
+        } else if (svgElements[i].tagName === 'path') {
+            var pathCoords = getPathCoords(svgElements[i].getAttribute('d'));
+            paths.push({
+                    'graph'     : parseInt(gId),
+                    'id_'       : '',
+                    'points'    : pathCoords,
+                    'fill'      : 'none',
+                    'stroke'    : '',
+                    'isRegion'  : false,
+                    'source'    : getClosestText(pathCoords[0], shapes),
```

Comment:
It may be that `getClosestText` returns undefined, when the source/target node doesn't have a label.

This is a larger problem to deal with later, but for now please just set this function to return the empty string rather than undefined (this will fix another parsing problem).